astrapy/db.py

```diff
@@ -694,6 +697,51 @@ def insert_many(
 
         return response
 
+    def batched_concurrent_insert_many(
+        self,
+        documents: List[API_DOC],
+        options: Optional[Dict[str, Any]] = None,
+        partial_failures_allowed: bool = False,
+        batch_size: int = MAX_INSERT_BATCH_SIZE,
+        concurrency: int = 1,
+    ) -> List[API_RESPONSE]:
```

Comment:
Would it be interesting/possible to merge the results into a single API_RESPONSE ?

Reply:
Ideally one could collate the `insertedIds` and the `errors` field from each, both expected to be lists if present (I go by memory with field names).
I wonder however if that would backfire by hardcoding some assumptions on what the API returns that in the future could theoretically change.

Reply:
OK.